### PR TITLE
Split template for all versions are yanked

### PR DESCRIPTION
**Pull request by aycabta**
This Pull Request is refreshed from https://github.com/rubygems/rubygems.org/pull/1408 what has some reverts in commit tree with discussion. This branch is cleared up by use of `git cherry-pick` minimal necessary commits.

[The new messages proposition](https://github.com/rubygems/rubygems.org/pull/1408#issuecomment-245435722) is applied to [other branch](https://github.com/aycabta/rubygems.org/commits/new-messages-for-all-versions-are-yanked).

/cc @indirect @sonalkr132

### CODE REVIEW

**Comment by sonalkr132** [comment]
Can you please use [partials](http://guides.rubyonrails.org/layouts_and_rendering.html#using-partials-to-simplify-views) to DRY the view. For example, [this section](https://github.com/aycabta/rubygems.org/blob/8b5681bd7113da0dbdce8bea59e21f6bfb8c2196/app/views/rubygems/show.html.erb#L56-L82) could easily be a `_gem_members` partial which can used in both `show` and `show_all_yanked` templates.

**Comment by aycabta** [comment]
If partials are applied to the views, it gets complex like https://github.com/rubygems/rubygems.org/pull/1408/commits/30c27e19b34164993157c0694c94e35deb05668d. The partials what commonalize some code such as `_gem_members` need more some conditional branches for `@latest_version` like [this block](https://github.com/rubygems/rubygems.org/pull/1408/commits/30c27e19b34164993157c0694c94e35deb05668d#diff-b434daadccccfc7b8638ca3e0f8db1f3R72). I splited off `show_all_yanked` because I want to bypass the complex of conditional branches.

File: app/controllers/rubygems_controller.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def index
   def show
     @latest_version = @rubygem.versions.most_recent
     @versions       = @rubygem.public_versions(5)
+    render :show_all_yanked if @rubygem.public_versions.count.zero?
```
Comment (simi) [inline app/controllers/rubygems_controller.rb]: I think `@rubygem.public_versions.none?` can be used here. And better to hide this into `Rubygem` model with some reasonable name so you can call something similar to `@rubygem.yanked?`.

*Is there actual any term for gem with all versions yanked? Does that mean that gem was deleted?*

File: app/views/rubygems/show_all_yanked.html.erb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<% @title = @rubygem.name %>
+<% @subtitle = @latest_version.try(:slug) %>
```
Comment (simi) [inline app/views/rubygems/show_all_yanked.html.erb]: Where are those used? Can we remove them?

Comment (aycabta) [inline app/views/rubygems/show_all_yanked.html.erb]: Those are used at [layout](https://github.com/aycabta/rubygems.org/blob/8b5681bd7113da0dbdce8bea59e21f6bfb8c2196/app/views/layouts/application.html.erb#L112-L125).

File: app/controllers/rubygems_controller.rb (hunk 1 above)
Comment (aycabta) [inline app/controllers/rubygems_controller.rb]: It's good idea.

> Does that mean that gem was deleted?

It just means all versions are yanked. The owner(s) can push new versions yet.

**Comment by aycabta** [comment]
https://github.com/rubygems/rubygems.org/pull/1423 are merged.

**Pull request closed without merging**